blueocean-dashboard/src/main/js/creation/ScmProvider.js

```diff
@@ -23,4 +23,11 @@ export default class ScmProvider {
         throw new Error('must implement getFlowManager');
     }
 
+    /**
+     * Called when the creation flow is about to end. Perform cleanup here.
+     */
+    destroyFlowManager() {
+        throw new Error('must implement destroyFlowManager');
+    }
+
```

Comment:
This would mainly be used for cleanup of timers or event listeners... similar to what is typically done in `componentWillUnmount`. I will add more details to the docs around its intended use. Is there anything else you'd like me to address on this, or concerns you have?